pkg/controller/siddhiprocess/deploymanager/application.go

```diff
@@ -0,0 +1,288 @@
+	corev1 "k8s.io/api/core/v1"
+
+	"gopkg.in/yaml.v2"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Application contains details about the siddhi app which need in the K8s deployment
+type Application struct {
+	Name               string                 `json:"appName"`
+	ContainerPorts     []corev1.ContainerPort `json:"containerPorts"`
+	Apps               map[string]string      `json:"apps"`
+	ServiceEnabled     bool                   `json:"serviceEnabled"`
+	PersistenceEnabled bool                   `json:"persistenceEnabled"`
+	Replicas           int32                  `json:"replicas"`
+}
+
+// Image holds the docker image details of the deployment
+type Image struct {
+	Name    string
+	Home    string
+	Secret  string
+	Profile string
+}
+
+// DeployManager creates and manage the deployment
+type DeployManager struct {
+	Application   Application
+	KubeClient    artifact.KubeClient
+	Labels        map[string]string
+	Image         Image
+	SiddhiProcess *siddhiv1alpha2.SiddhiProcess
+}
+
+// SPConfig contains the state persistence configs
+type SPConfig struct {
+	Location string `yaml:"location"`
+}
+
+// StatePersistence contains the StatePersistence config block
+type StatePersistence struct {
+	SPConfig SPConfig `yaml:"config"`
+}
+
+// Config contains the siddhi config block
+type Config struct {
+	StatePersistence StatePersistence `yaml:"state.persistence"`
+}
+
+// CreateLabels creates labels for each deployment
+func (d *DeployManager) CreateLabels() {
+	labels := map[string]string{
+		"siddhi.io/name":     CRDName,
+		"siddhi.io/instance": d.Application.Name,
+		"siddhi.io/version":  OperatorVersion,
+		"siddhi.io/part-of":  OperatorName,
+	}
+	d.Labels = labels
+}
+
+// Deploy creates a deployment according to the given Application specs.
+// It create and mount volumes, create config maps, populates envs that needs to the deployment.
+func (d *DeployManager) Deploy() (operationResult controllerutil.OperationResult, err error) {
+
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	var imagePullSecrets []corev1.LocalObjectReference
+	var depStrategy appsv1.DeploymentStrategy
+	configParameter := ""
+	appParameter := ""
+	appsMap := make(map[string]string)
+
+	d.CreateLabels()
+	containerPorts := d.Application.ContainerPorts
+	if d.Image.Secret != "" {
+		secret := createLocalObjectReference(d.Image.Secret)
+		imagePullSecrets = append(imagePullSecrets, secret)
+	}
+
+	q := siddhiv1alpha2.PV{}
+	if d.Application.PersistenceEnabled {
+		if !d.SiddhiProcess.Spec.PV.Equals(&q) {
+			pvcName := d.Application.Name + PVCExtension
+			err = d.KubeClient.CreateOrUpdatePVC(
+				pvcName,
+				d.SiddhiProcess.Namespace,
+				d.SiddhiProcess.Spec.PV.AccessModes,
+				d.SiddhiProcess.Spec.PV.Resources.Requests.Storage,
+				d.SiddhiProcess.Spec.PV.Class,
+				d.SiddhiProcess,
+			)
+			if err != nil {
+				return
+			}
+			mountPath := ""
```

Comment:
You don't need to initialize `mountPath` separately. You can assign and initialize in the below line.

Reply:
Fixed in https://github.com/siddhi-io/siddhi-operator/pull/123/commits/99c34b77761500db9edbf37569a84caa35738477